packages/glimmer-runtime/lib/builder.ts

```diff
@@ -193,16 +188,35 @@ export class ElementStack implements Cursor {
 
   flushElement() {
     let parent  = this.element;
-    let element = this.element = this.constructing;
+    let element = this.constructing;
 
     this.dom.insertBefore(parent, element, this.nextSibling);
 
     this.constructing = null;
     this.operations = null;
-    this.nextSibling = null;
+
+    this.pushElement(element);
+    this.blockStack.current.openElement(element);
+  }
+
+  pushRemoteElement(element: Simple.Element) {
+    this.pushElement(element);
+
+    let tracker = new RemoteBlockTracker(element);
+    this.pushBlockTracker(tracker, true);
+  }
+
+  popRemoteElement() {
+    this.popBlock();
+    this.popElement();
+  }
+
+  private pushElement(element: Simple.Element) {
+    this.element = element;
     this.elementStack.push(element);
+
+    this.nextSibling = null;
     this.nextSiblingStack.push(null);
```

Comment:
we can make this configurable for bonus points, which would allow inserting into the middle of an element :wink:

but we can do it as a follow-up.